backend/src/meme/controllers/meme-controller.ts

```diff
@@ -0,0 +1,109 @@
+import {
+	Body,
+	Controller,
+	Get,
+	HttpException,
+	HttpStatus,
+	Param,
+	ParseIntPipe,
+	Post,
+	UploadedFile,
+	UseGuards,
+	UseInterceptors
+} from "@nestjs/common";
+import { JwtAuthGuard } from "../../auth/guards/jwt-auth.guard";
+import { MemeService } from "../services/meme.service";
+import { FileInterceptor } from "@nestjs/platform-express";
+import { MemeDto } from "../model/dto/meme.dto";
+@Controller("meme")
+class MemeController {
+	constructor(private memeService: MemeService) {}
+
+	@UseGuards(JwtAuthGuard)
+	@Get("random")
+	async getRandomMeme() {
+		const file = await this.memeService.findRandom();
+
+		if (!file) {
+			throw new HttpException("Not found", HttpStatus.NOT_FOUND);
```

Comment:
Use predefined Exception from Nest ep. here u can use "NotFoundException".

Refactor the rest of youre Exceptions.

https://docs.nestjs.com/exception-filters#exception-filters